store/project_member.go

```
@@ -123,70 +123,120 @@ func (s *ProjectMemberService) DeleteProjectMember(ctx context.Context, delete *
 	return nil
 }
 
-// SetProjectMember set the project member with provided project member list
-func (s *ProjectMemberService) SetProjectMember(ctx context.Context, set *api.ProjectMemberSet) (createdMember, deletedMember []*api.ProjectMember, err error) {
+// getBatchUpdatePrincipalIDList return the principal ID for each operation (this function may be a litter overhead, but it is easy to be tested)
+func getBatchUpdatePrincipalIDList(oldPrincipalIDList []int, newPrincipalIDList []int) (createPrincipalIDList, patchPrincipalIDList, deletePrincipalIDList []int, err error) {
```

Do we have a case that we need to return an error?